rotkehlchen/exchanges/bybit.py

```diff
@@ -0,0 +1,514 @@
+        changed = super().edit_exchange_credentials(credentials)
+        if changed is True:
+            self.session.headers.update({'X-BAPI-API-KEY': self.api_key})
+
+        return changed
+
+    def first_connection(self) -> None:
+        if self.first_connection_made is True:
+            return
+
+        try:
+            result = self._api_query(path='user/query-api')
+        except RemoteError as e:
+            log.error(f'Could not get api key information at connection for bybit due to {e}')
+            return
+
+        try:
+            self.is_unified_account = result['uta'] == 1
+        except KeyError:
+            log.error('Failed to read uta value from first connection due to key error')
+            return
+
+        self.first_connection_made = True
+
+    def _generate_signature(self, param_str: str) -> str:
+        signature_hash = hmac.new(
+            key=self.secret,
+            msg=param_str.encode('utf-8'),
+            digestmod=hashlib.sha256,
+        )
+        return signature_hash.hexdigest()
+
+    def _api_query(
+            self,
+            path: Literal[
+                'account/wallet-balance',
+                'account/transaction-log',
+                'order/history',
+                'user/query-api',
+                'asset/deposit/query-record',
+                'asset/withdraw/query-record',
+            ],
+            options: dict | None = None,
+    ) -> dict:
+        """
+        Query bybit endpoints.
+        May raise:
+        - RemoteError
+        """
+        url = f'{self.uri}/{path}'
+        timeout = CachedSettings().get_timeout_tuple()
+        tries = CachedSettings().get_query_retry_limit()
+        requires_auth = path in self.authenticated_methods
+
+        while True:
+            log.debug('Bybit API Query', url=url, options=options)
+            if requires_auth:
+                timestamp = ts_now_in_ms()
+                # the order in this string is defined by the api
+                param_str = str(timestamp) + self.api_key + self.recv_window
+                if options is not None:
+
+                    options = dict(sorted(options.items()))
+                    param_str += '&'.join(  # params need to be sorted to be correctly validated
+                        [
+                            str(k) + '=' + str(v)
+                            for k, v in sorted(options.items())
+                            if v is not None
+                        ],
+                    )
+
+                signature = self._generate_signature(param_str=param_str)
+                self.session.headers = {
+                    'Content-Type': 'application/json',
+                    'X-BAPI-SIGN-TYPE': '2',
+                    'X-BAPI-RECV-WINDOW': self.recv_window,
+                    'X-BAPI-API-KEY': self.api_key,
+                    'X-BAPI-TIMESTAMP': str(timestamp),
+                    'X-BAPI-SIGN': signature,
+                }
```

Comment:
HEre you are adding many constants. Don't. The only thing you need to add is the signature and th etimestamp. Everything else is constant and should be added at the constructor.